src/commands.ts

```diff
@@ -0,0 +1,97 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import * as Utils from './utils';
+import { DotNetInfo, requireDotNetSdk, runDotNetCommand, findProjectTemplate } from './dotnet';
+import { CommandObserver } from './CommandObserver';
+
+export default function registerCommands(commandObserver: CommandObserver, packageInfo: Utils.IPackageInfo): vscode.Disposable[] {
+    let dotNetSdkVersion = packageInfo.requiredDotNetCoreSDK;
+    return [
+        vscode.commands.registerCommand('pgproj.build.all', async () => {
+            requireDotNetSdk(dotNetSdkVersion).then(
+                dotnet => {
+                    buildAllProjects(dotnet, commandObserver);
+            });
+        }),
+        vscode.commands.registerCommand('pgproj.build.current', async (args) => {
+            requireDotNetSdk(dotNetSdkVersion).then(
+                dotnet => {
+                    buildCurrentProject(args, dotnet, commandObserver);
+            });
+        }),
+        vscode.commands.registerCommand('pgproj.add.new', async (args) => {
+            requireDotNetSdk(dotNetSdkVersion).then(
+                dotnet => {
+                    addNewPostgreSQLProject(args, dotnet, packageInfo.projectTemplateNugetId, commandObserver);
+            });
+        }),
+    ];
+}
+
+function buildCurrentProject(args, dotNetSdk: DotNetInfo, commandObserver: CommandObserver) {
+    let project = '';
+    if (!args) {
+        const activeEditor = vscode.window.activeTextEditor;
+        if (activeEditor !== undefined) {
+            project = activeEditor.document.uri.fsPath;
+        }
+    } else {
+        project = args.fsPath;
+    }
+
+    commandObserver.clear();
+    dotnetBuild(dotNetSdk, project, commandObserver);
+}
+
+async function buildAllProjects(dotNetSdk: DotNetInfo, commandObserver: CommandObserver): Promise<void> {
+    try {
+        let projects = await vscode.workspace.findFiles('{**/*.pgproj}');
+        commandObserver.clear();
+        for (let project of projects) {
+            dotnetBuild(dotNetSdk, project.fsPath, commandObserver);
+        }
+    } catch (err) {
+        vscode.window.showInformationMessage(err);
+    }
+}
+
+function dotnetBuild(dotNetSdk: DotNetInfo, project: string, commandObserver: CommandObserver): Promise<void> {
+    return new Promise<void>((_resolve, _reject) => {
+        let args = ['build', project];
+        runDotNetCommand(dotNetSdk, args, commandObserver);
+    });
+}
+
+function addNewPostgreSQLProject(args, dotNetSdk: DotNetInfo, projectTemplateNugedId: string, commandObserver: CommandObserver) {
+    let folder = args.fsPath;
```

Comment:
Btw, what will be this folder? Is it going to create a folder per project or all projects will be created in just the current folder? I'm thinking if we should actually create project folder and park ```pgproj``` in there. What will be the name of the ```pgproj``` file? Is it going to prompt the user as part of ```dotnet new```? Is it going to take care of subfolders too?

Reply:
This is the folder that user select to create the pgproject in. We only allow them to right click on a folder and do Add new project. By default it takes the name of the folder. I am trying to see if user can provide a name to the command that we can pass, then we will create the pgproj with that name. Right now you can do it via command line. haven't added the naming in ADS yet.

Reply:
https://code.visualstudio.com/api/references/vscode-api - ```showInputBox``` is what you want, probably.

Reply:
yep, using that.